spring-boot-project/spring-boot-actuator-autoconfigure/src/main/java/org/springframework/boot/actuate/autoconfigure/jdbc/DataSourceHealthContributorAutoConfiguration.java

```diff
@@ -64,8 +71,7 @@
 @ConditionalOnEnabledHealthIndicator("db")
 @AutoConfigureAfter(DataSourceAutoConfiguration.class)
 @EnableConfigurationProperties(DataSourceHealthIndicatorProperties.class)
-public class DataSourceHealthContributorAutoConfiguration extends
-		CompositeHealthContributorConfiguration<AbstractHealthIndicator, DataSource> implements InitializingBean {
```

Comment:
I broke out of the CHCC hierarchy as this stretched that contract a bit too much. We need the `createIndicator` to return `HealthContributor` rather than `AbstractHealthIndicator` to allow for the case where one of the datasource beans is actual the AbstractRoutingDataSource which in turn is composite contributor.